libs/vkd3d/utils.c

```diff
@@ -647,6 +648,34 @@ VkFormat vkd3d_get_vk_format(DXGI_FORMAT format)
     return VK_FORMAT_UNDEFINED;
 }
 
+/* get some size-based low bits for prioritization in the same way as d3d12 */
+inline uint32_t vkd3d_get_priority_adjust(VkDeviceSize size)
+{
+    return min((size / (10 * 1048576)), 0xFFFFUL);
+}
+
+/* map from 32-bit d3d prio to float (0..1) vk prio.*/
+float vkd3d_convert_to_vk_prio(D3D12_RESIDENCY_PRIORITY d3d12prio)
+{
+    float result;
+
+    /* align D3D12_RESIDENCY_PRIORITY_NORMAL (the default d3d12 prio) with
+       0.5 (the default vk prio) so neither kind wins without explicit prio */
+    if (d3d12prio <= D3D12_RESIDENCY_PRIORITY_NORMAL)
+        result = 0.5f * ((double)d3d12prio) / D3D12_RESIDENCY_PRIORITY_NORMAL;
```

Comment:
Weird mixing of float and double.

Reply:
Is it the use of double you object to or using 0.5f instead of 0.5?  Yeah at least the latter is weird - can fix.

Reply:
The math should be done in float I think. The end-result is float either way.